pkg/controller/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -91,7 +91,7 @@ func (r *ReconcileUserSignupCleanup) Reconcile(request reconcile.Request) (recon
 	}
 	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
 
-	if states.VerificationRequired(instance) && !instance.Spec.Approved {
+	if states.VerificationRequired(instance) && !states.Approved(instance) {
```

Comment:
This is a different controller and we can't guarantee that the UserSignup is already migrated by the usersignup controller. We should either add a migration step here too or check both `states.Approved()` and `instance.Spec.Approved` in case it's not migrated yet.

Reply:
The migration block has been added to the UserSignup cleanup controller also, and tests added for both controllers.

Reply:
But this controller should not contain any migration - the migration should be only in `usersignup_controller.go`.
@alexeykazakov proposed modifying the `if` statement, so it would check both the `states.Approved()` as well as  `instance.Spec.Approved` and not copy-pasting the migration code.

Reply:
The if check won't work here, the migration needs to set the approved spec property to false otherwise it breaks our tests.

Reply:
Why? What's wrong on:
```
if states.VerificationRequired(instance) && !(states.Approved(instance) || instance.Spec.Approved)
```
?

Reply:
Changes made as suggested.